bigtable/tests/unit/test_client.py

```diff
@@ -86,6 +212,19 @@ def test_constructor_both_admin_and_read_only(self):
                 project=self.PROJECT, credentials=credentials,
                 admin=True, read_only=True)
 
+    def test_constructor_with_emulator_host(self):
+        import grpc
+        credentials = _make_credentials()
+        emulator_host = "localhost:8081"
+        with mock.patch('os.getenv') as mocked:
+            mocked.return_value = emulator_host
+            client = self._make_one(project=self.PROJECT,
+                                    credentials=credentials)
+
+        self.assertEqual(client._emulator_host, emulator_host)
+        self.assertIsInstance(client._emulator_channel,
+                              grpc._channel.Channel)
```

Comment:
This really wants an assertion that `emulator_host` was correctly passed into the `grpc.insecure_channel` factory, which we could mock out via:

```python
    def test_constructor_with_emulator_host(self):
        import grpc
        credentials = _make_credentials()
        emulator_host = "localhost:8081"
        with mock.patch('os.getenv') as mocked:
            with mock.patch('grpc.insecure_channel') as factory:
                mocked.return_value = emulator_host
                client = self._make_one(
                    project=self.PROJECT, credentials=credentials)

        self.assertEqual(client._emulator_host, emulator_host)
        self.assertIsInstance(client._emulator_channel,
                              grpc._channel.Channel)
        factory.assert_called_once_with(emulator_host)
```

Reply:
6b0abee04bd8a043025c8b1ba116a3bf80d98d49